packages/channel-client/tests/fakes/fake-channel-provider.ts

```diff
@@ -0,0 +1,247 @@
+import {ChannelProviderInterface} from '@statechannels/channel-provider/src';
```

Comment:
`FakeChannelProvider` should definitely be `channel-provider` though because of the types used across the files, from an initial stab at moving the code around it seems like a slightly larger refactor of type usage needs to take place first (for example there's a discrepancy between the `ChannelResult` type being referenced in the `ChannelClient`, and the interface definition, and the `ChannelResult` type that's defined in the `client-api-schema` package that should be the common reference for all package working with that type)

Once these sort of discrepancies are solved we can cleanly move `FakeChannelProvider` to the right place. The current placement prioritized getting the fake working.